Plotter/python/sample/HistSet.py

```diff
@@ -1,130 +1,195 @@
-    ndata = len(self.vars) if isinstance(self.vars,dict) else 1
-    print(">>> HistSet: nvars=%d, ndata=%d, nexp=%d, nsig=%d"%(
-                        nvars,ndata,len(self.exp),len(self.signal)))
-    if full:
-      print(">>>   vars=%s"%(self.vars))
-      print(">>>   data=%s"%(self.data))
-      print(">>>   exp=%s"%(self.exp))
-      print(">>>   sig=%s"%(self.signal))
-    else:
-      print(">>>   vars = %s"%((', '.join(repr(str(v)) for v in self.vars) if isinstance(self.vars,list) else self.vars)))
-      if self.data and isinstance(self.data,dict):
-        print(">>>   data = { %s }"%(', '.join(repr(h.GetName()) for v,h in self.data.items())))
-      elif isinstance(self.data,TH1):
-        print(">>>   data = %r"%(self.data.GetName()))
-      else:
-        print(">>>   data = %s"%(self.data))
-      def printset(dtype,set):
-        """Help function to print exp and signal histogram sets (dictionaries or lists)."""
-        if isinstance(set,dict):
-          for var, hlist in set.items():
-            print(">>>   %s['%s'] = [ %s ]"%(dtype,var,', '.join(repr(h.GetName()) for h in hlist)))
-        elif isinstance(set,list) and set:
-          print(">>>   %-4s = [ %s ]"%(dtype,', '.join(repr(h.GetName()) for h in set)))
+  def results(self,singlevar=False,singlesel=False,popvar=None):
+    """Return simple nested dictionaries. { selection: { variable: HistSet } }
+    Convert for just a single variable (issinglevar==True), and/or single selection (issinglesel==True)."""
+    results = self._dict
+    if popvar!=None: # remove variable from dictionary (used by Sample.getmean)
+      for sel in results:
+        results[sel].pop(popvar,None)
+    if singlevar: # convert result to { selection: HistSet }
+      for sel in results.keys():
+        if len(results[sel])>=2:
+          LOG.warn("HistDict.results: singlevar=%r, singlesel=%r, but found more than one selection key in self._dict=%r"%(
+            singlevar,singlesel,self._dict))
+        if len(results[sel])==0: # no variables/hists
+          results.pop(sel) # delete nested dictionary
         else:
-          print(">>>   %-4s = %s"%(dtype,set))
-      printset('exp',self.exp)
-      printset('sig',self.signal)
-  
+          varkey = list(results[sel].keys())[0] # get first (and only?) key
+          results[sel] = results[sel][varkey]
+    if singlesel: # convert result to { variable: HistSet } or if singlevar==True: a single HistSet
+      if len(results)>=2:
+        LOG.warn("HistDict.results: singlevar=%r, singlesel=%r, but found more than one selection key in self._dict=%r"%(
+          singlevar,singlesel,self._dict))
+      if len(results)==0: # no selections
+        results = HistSet() # empty hist set
+      else:
+        selkey  = list(results.keys())[0] # get first (and only?) key
+        results = results[selkey] # get single nested dictionary or HistSet
+    return results
 
 
+from TauFW.Plotter.sample.utils import LOG
```

Comment:
Is there a specific reason why this instruction is at the end of the code? @IzaakWN @pmastrap

Reply:
To avoid circular import: `HistSet` is imported into `TauFW.Plotter.sample.utils`.